memebot/memebot.py

```diff
@@ -116,56 +120,57 @@ async def setup_voice(self):
 
             self.guild_player_config[guild.id] = PlayerConfig(
                 player=player,
-                voice_channel_id=config.voice_channel_id,
-                text_channel_id=config.text_channel_id)
+                voice_channel=voice_channel,
+                text_channel=text_channel)
 
         log.info('done setting up voice channels')
 
 
 PlayerConfigProvider = Callable[[int], Optional[PlayerConfig]]
-PlayerCommandExecutor = Callable[[str, discord.TextChannel, PlayerConfig], Awaitable[None]]
+PlayerCommandExecutor = Callable[[Command, discord.TextChannel, PlayerConfig], Awaitable[None]]
 
 
-def ignore_by_player_config_provider(player_config_provider: PlayerConfigProvider) -> Callable[[PlayerCommandExecutor], CommandExecutor]:
+def filter_commands(*, player_config_provider: PlayerConfigProvider, voice_channel_restricted: bool) -> Callable[[PlayerCommandExecutor], CommandExecutor]:
     """Decorator factory to ignore player commands on unsupported servers/channels
     Arguments:
         player_config_provider {Provider[PlayerConfig]} -- provides Player and PlayerConfig
             based on received message's guild id
+        voice_channel_restricted {bool} -- if true, will ignore commands by users not connected
+            to voice channel
     Returns:
         CommandExecutor -- That will execute the PlayerCommandExecutor when a message
             is received in the allowed channel
     """
-    async def no_op_awaitable():
-        # TODO: better feedback
-        pass
-
     def wrapper(func: PlayerCommandExecutor):
         @executor()
         @wraps(func)
-        def wrapped(command_arg: str, channel: discord.TextChannel):
+        def wrapped(command: Command, channel: discord.TextChannel):
             config = player_config_provider(channel.guild.id)
             if not config:
                 log.debug(f'could not find config for guild id {channel.guild.id}')
-                return no_op_awaitable()
-            if config.text_channel_id != channel.id:
-                log.debug(f'player command in guild {channel.guild.id} did not come from PlayerConfig.text_channel_id {config.text_channel_id}')
-                return no_op_awaitable()
-            return func(command_arg, channel, config)
+                return channel.send(f'Player not configured for this server')
+            if config.text_channel.id != channel.id:
+                log.debug(f'player command in guild {channel.guild.id} did not come from PlayerConfig.text_channel_id {config.text_channel.id}')
+                return channel.send(f'Command {command.raw_command!r} must be made in #{config.text_channel.name}')
```

Comment:
Confused on this. Noticing that there is no boolean check to see if something is channel restricted. I'm guessing you have determined that you will only want audio commands to be issued from one channel at a time, which makes sense, but you may want this to be dynamic, aka, setting can be set by admin as opposed to hardcoded. Could be that I'm a dumb-dumb and misunderstanding tho

Reply:
The flag i added only checks if you're on the voice channel. I supposed I should make this more explicit now that this is named as if it works for any commands. Maybe `filter_player_commands` is more appropriate? ATM there isn't any use case to have a `text_channel_restricted` (since this would always be true), so I would hold off adding it for now.